File: leetcode/src/test/java/org/luvx/leetcode/java/medium/_725/SolutionTest.java
package org.luvx.leetcode.java.medium._725;

import org.junit.Test;
import org.luvx.leetcode.link.ListNode;
import org.luvx.leetcode.util.LinkUtils;

public class SolutionTest {
    @Test
    public void splitListToParts() {
        Solution exec = new Solution();
        for (int i = 1; i < 15; i++) {
            ListNode h = LinkUtils.initLinked(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
            ListNode[] listNodes = exec.splitListToParts(h, i);
            for (ListNode h1 : listNodes) {
                LinkUtils.printLink(h1);
            }
        }
    }
}

File: leetcode/src/test/kotlin/org/luvx/leetcode/java/medium/_429/SolutionTest.kt
package org.luvx.leetcode.java.medium._429;

import org.luvx.leetcode.tree.Node
import org.luvx.leetcode.out
import kotlin.test.BeforeTest
import kotlin.test.Test

class SolutionTest {
    private var root: Node? = null

    @BeforeTest
    fun before() {
        root = Node(
            5,
            listOf(
                Node(
                    4,
                    listOf(
                        Node(
                            11,
                            listOf(
                                Node(7),
                                Node(2),
                                Node(4)
                            )
                        )
                    )
                ),
                Node(
                    8,
                    listOf(
                        Node(13),
                        Node(
                            4,
                            listOf(
                                Node(1)
                            )
                        )
                    )
                )
            )
        )
    }

    /**
     * <pre>
     *          5
     *         / \
     *        4   8
     *       /   / \
     *      11  13  4
     *     /| \      \
     *    7 2  4      1
     * </pre>
     */
    @Test
    fun levelOrderTest() {
        val exec = Solution()
        exec.levelOrder(root).out()
    }
}